src/clr-angular/data/datagrid/datagrid-column-toggle-button.spec.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Component, QueryList, TemplateRef, ViewChildren} from "@angular/core";
+
+import {ClrDatagridColumnToggle} from "./datagrid-column-toggle";
+import {ClrDatagridColumnToggleButton} from "./datagrid-column-toggle-button";
+import {DatagridHideableColumnModel} from "./datagrid-hideable-column.model";
+import {TestContext} from "./helpers.spec";
+import {ColumnToggleButtonsService} from "./providers/column-toggle-buttons.service";
+import {HideableColumnService} from "./providers/hideable-column.service";
+
+export default function(): void {
+    describe("Datagrid Column Toggle Button component", function() {
+        describe("Typescript API", function() {
+            let columnToggleButtons: ColumnToggleButtonsService;
+            let component: ClrDatagridColumnToggleButton;
+
+            beforeEach(function() {
+                columnToggleButtons = new ColumnToggleButtonsService();
+                component = new ClrDatagridColumnToggleButton(columnToggleButtons);
+            });
+
+            it("knows if the button isOk()", function() {
+                component.type = "ok";
+                expect(component.isOk()).toBe(true);
+                component.type = "selectAll";
+                expect(component.isOk()).toBe(false);
+            });
+
+            it("gets the correct classes", function() {
+                component.type = "ok";
+                expect(component.getClasses()).toEqual("btn btn-primary");
+                component.type = "selectAll";
+                expect(component.getClasses()).toEqual("btn btn-sm btn-link p6 text-uppercase");
```

Comment:
Nitpick, but this feels like a view test. You're pretty much testing that the correct CSS classes are added to the host.

Reply:
Its on the fence, because I have a class method that returns these values. Its implicitly covered in the view "it has a button" test by checking the className there, but felt like this was useful to check that we didn't change the classes here accidentally.